build/tasks/minify.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+
+const UglifyJS = require( "uglify-js" );
+const fs = require( "fs" );
+const path = require( "path" );
+const processForDist = require( "./dist" );
+const getTimestamp = require( "./lib/getTimestamp" );
+
+const rjs = /\.js$/;
+
+module.exports = async function minify( { dir, filename } ) {
+	const filepath = path.join( dir, filename );
+	const contents = await fs.promises.readFile( filepath, "utf8" );
+	const version = /jQuery JavaScript Library ([^\n]+)/.exec( contents )[ 1 ];
+	const banner = `/*! jQuery ${version}` +
+		" | (c) OpenJS Foundation and other contributors" +
+		" | jquery.org/license */";
+
+	const { code, error, map: incompleteMap, warning } = UglifyJS.minify(
+		contents,
+		{
+			compress: {
+				hoist_funs: false,
+				loops: false,
+
+				// Support: IE <11
+				// typeofs transformation is unsafe for IE9-10
+				// See https://github.com/mishoo/UglifyJS2/issues/2198
+				typeofs: false
+			},
+			output: {
+				ascii_only: true,
+
+				// Support: Android 4.0 only
+				// UglifyJS 3 breaks Android 4.0 if this option is not enabled.
+				// This is in lieu of setting ie for all of mangle, compress, and output
+				ie8: true,
+				preamble: banner
+			},
+			sourceMap: {
+				filename
+			}
+		}
+	);
+
+	if ( error ) {
+		throw new Error( error );
+	}
+
+	if ( warning ) {
+		console.warn( warning );
+	}
+
+	const minFilename = filename.replace( rjs, ".min.js" );
+	const mapFilename = filename.replace( rjs, ".min.map" );
+
+	// The map's `files` & `sources` property are set incorrectly, fix
+	// them via overrides from the task config.
+	const map = JSON.stringify( {
+		...JSON.parse( incompleteMap ),
+		file: minFilename,
+		sources: [ filename ]
+	} );
```

Comment:
This is a problem on `main` due to some SWC bugs, we shouldn't need to adjust manually here.

Also, can you upload the generated source map at https://sokra.github.io/source-map-visualization/ and do a basic check that it makes sense?

Reply:
While the filename is fine, it's still needed because the source is unknown. I can update the comment. Checked in the source map visualizer, though.